### PR TITLE
Use goroutines for storing POI.place and its geo-location in Redis

### PR DESCRIPTION
**Pull request by rwangsc18**
## Description
SetPlacesOnCategory() was not effectively using goroutines

## Solution

- Rename SetPlacesOnCategory to SetPlacesAddGeoLocations
- Add worker function for adding Geo locations for places

## Testing
- [ ] Integration testing on Heroku staging
- [ ] Added new unit tests

## Checks
- [x] Have you removed commented code?
- [x] Have you used gofmt to format your code?

### CODE REVIEW

File: iowrappers/redis_client.go [hunk 1]
```diff
@@ -151,26 +151,35 @@ func (r *RedisClient) StorePlacesForLocation(context context.Context, geocodeInS
...
+// keys are identified with different place categories
+func (r *RedisClient) GeoAddWorkerByCategory(c context.Context, place POI.Place, wg *sync.WaitGroup) {
+	defer wg.Done()
+	placeCategory := POI.GetPlaceCategory(place.LocationType)
+	geoLocation := &redis.GeoLocation{
+		Name:      place.ID,
+		Latitude:  place.GetLocation().Latitude,
+		Longitude: place.GetLocation().Longitude,
+	}
+	redisKey := "placeIDs:" + strings.ToLower(string(placeCategory))
```
Comment (rwangsc18) [inline iowrappers/redis_client.go:178]: @timwangmusic  is this the part you were mentioning about increasing the number of buckets we store geolocations? 

Currently we just split by visit and eatory IIUC. this leads to an issue where the retrieved results for restaurants can have limited number of "$$$" ones.

Comment (timwangmusic) [reply iowrappers/redis_client.go:178]: This is not the part I mentioned. The matching package requests the number of results for a category, see `NearbySearchForCategory` method.

File: iowrappers/redis_client.go [hunk 2]
```diff
@@ -151,26 +151,35 @@ func (r *RedisClient) StorePlacesForLocation(context context.Context, geocodeInS
...
-		utils.LogErrorWithLevel(cmdErr, utils.LogError)
-
-		r.setPlace(context, place, wg)
+		go r.GeoAddWorkerByCategory(c, place, wg)
+		go r.setPlace(c, place, wg)
 	}
 	wg.Wait()
 }
 
+// GeoAddWorkerByCategory adds geo-locaiton of the place,
```
Comment (timwangmusic) [inline iowrappers/redis_client.go:168]: wrong spelling of location

**Review by timwangmusic — APPROVED** [review]
LGTM

File: iowrappers/redis_client.go (hunk 1 above)
Comment (rwangsc18) [reply iowrappers/redis_client.go:178]: sg, thanks!

**Pull request merged**